spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -2,6 +2,14 @@
 
 RSpec.describe Broadcasts::WelcomeNotification::Generator, type: :service do
   describe "::call" do
+    let(:receiving_user) { create(:user) }
+    let(:user) { create(:user) }
+    let!(:welcome_broadcast) { create(:welcome_broadcast, :active) }
+
+    before do
+      allow(User).to receive(:mascot_account).and_return(create(:user))
```

Comment:
I was wondering if we need a brand new user object before each test since it's the result of a mocked function. Maybe we can just cache that "mascot_account" in another `let()` statement and replace `create(:user)` here with it. Haven't tried it, maybe you actually new a fresh account :D

Reply:
This is a great suggestion and after trying it, it seems to work as expected! 😄